CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,415 @@
+                    ThemeID = 12
+                }
+            };
+
+            _studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(1)).ReturnsAsync(true);
+
+            _mentorRepositoryMock.Setup(x => x.IsEntityExistAsync(2)).ReturnsAsync(true);
+
+            _themeRepositoryMock.Setup(x => x.IsEntityExistAsync(12)).ReturnsAsync(true);
+
+            Initialize(createScheduleDto);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.CreateScheduleAsync(createScheduleDto);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_NotValidDaysOfWeekInRelativeMonthly_ShouldReturnValidationError()
+        {
+            //Arrange
+            var createScheduleDto = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.AbsoluteMonthly,
+                    Interval = 1
+                },
+
+                Range = new OccurenceRange { },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = 1,
+                    MentorID = 2,
+                    ThemeID = 12
+                }
+            };
+
+            _studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(1)).ReturnsAsync(true);
+
+            _mentorRepositoryMock.Setup(x => x.IsEntityExistAsync(2)).ReturnsAsync(true);
+
+            _themeRepositoryMock.Setup(x => x.IsEntityExistAsync(12)).ReturnsAsync(true);
+
+            Initialize(createScheduleDto);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.CreateScheduleAsync(createScheduleDto);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_NullRequest_ShouldReturnValidationError()
+        {
+            //Arrange
+            CreateScheduleDto createScheduleDto = null;
+
+            Initialize(createScheduleDto);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.CreateScheduleAsync(createScheduleDto);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        protected override Mock<IUnitOfWork> GetUnitOfWorkMock()
+        {
+            var mock = new Mock<IUnitOfWork>();
+            return mock;
+        }
```

Comment:
Why override this method and then implement it in exactly same way as in the base class?

Reply:
Ok, removed